source/script_menu.cpp

```diff
@@ -496,11 +496,16 @@ ResultType UserMenu::AddItem(LPTSTR aName, UINT aMenuID, IObject *aCallback, Use
 	else
 		name_dynamic = Var::sEmptyString; // So that it can be detected as a non-allocated empty string.
 	UserMenuItem *menu_item = new UserMenuItem(name_dynamic, length + 1, aMenuID, aCallback, aSubmenu, this);
-	if (!menu_item) // Should also be very rare.
+	if ( !menu_item // Should also be very rare.
+		 || (*aOptions && !UpdateOptions(menu_item, aOptions)) ) // update options
 	{
 		if (name_dynamic != Var::sEmptyString)
 			free(name_dynamic);
-		return g_script.ScriptError(ERR_OUTOFMEM);
+		if (menu_item)
+			delete menu_item; 
```

Comment:
Calling the UserMenuItem destructor (via delete) after calling `free(name_dynamic)` is not safe. We must `free(name_dynamic)` when and only when `!menu_item`. Commit 7a7e17af fixes this.

Reply:
I see my mistake, thanks.